src/core/radio/ble_secure.cpp

```diff
@@ -0,0 +1,540 @@
+    {
+        mTls.Disconnect();
+    }
+
+    if (mBleConnectionOpen)
+    {
+        otPlatBleGapDisconnect(&GetInstance());
+    }
+}
+
+void BleSecure::SetPsk(const MeshCoP::JoinerPskd &aPskd)
+{
+    static_assert(static_cast<uint16_t>(MeshCoP::JoinerPskd::kMaxLength) <=
+                      static_cast<uint16_t>(MeshCoP::Dtls::kPskMaxLength),
+                  "The maximum length of TLS PSK is smaller than joiner PSKd");
+
+    SuccessOrAssert(mTls.SetPsk(reinterpret_cast<const uint8_t *>(aPskd.GetAsCString()), aPskd.GetLength()));
+}
+
+#if defined(MBEDTLS_BASE64_C) && defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+Error BleSecure::GetPeerCertificateBase64(unsigned char *aPeerCert, size_t *aCertLength)
+{
+    Error error;
+
+    VerifyOrExit(aCertLength != nullptr, error = kErrorInvalidArgs);
+
+    error = mTls.GetPeerCertificateBase64(aPeerCert, aCertLength, *aCertLength);
+
+exit:
+    return error;
+}
+#endif // defined(MBEDTLS_BASE64_C) && defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+
+Error BleSecure::SendMessage(ot::Message &aMessage)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsConnected(), error = kErrorInvalidState);
+    VerifyOrExit(mSentMessage != nullptr, error = kErrorNoBufs);
+    SuccessOrExit(error = mSentMessage->AppendBytesFromMessage(aMessage, 0, aMessage.GetLength()));
+
+    if (mSentMessage->GetLength() > kFlushThreshold)
+    {
+        error = Flush();
+    }
+
+    if (error == kErrorNone)
+    {
+        aMessage.Free();
+    }
+
+exit:
+    return error;
+}
+
+Error BleSecure::Send(uint8_t *aBuf, uint16_t aLength)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsConnected(), error = kErrorInvalidState);
+    VerifyOrExit(mSentMessage != nullptr, error = kErrorNoBufs);
+    SuccessOrExit(error = mSentMessage->AppendBytes(aBuf, aLength));
+
+    if (mSentMessage->GetLength() > kFlushThreshold)
+    {
+        error = Flush();
+    }
+
+exit:
+    return error;
+}
+
+Error BleSecure::SendApplicationTlv(uint8_t *aBuf, uint16_t aLength)
+{
+    if (aLength > 254)
+    {
+        ot::ExtendedTlv tlv;
+
+        tlv.SetType(ot::MeshCoP::TcatAgent::kTlvSendApplicationData);
+        tlv.SetLength(aLength);
+        Send(reinterpret_cast<uint8_t *>(&tlv), sizeof(tlv));
+    }
+    else
+    {
+        ot::Tlv tlv;
+
+        tlv.SetType(ot::MeshCoP::TcatAgent::kTlvSendApplicationData);
+        tlv.SetLength(aLength);
+        Send(reinterpret_cast<uint8_t *>(&tlv), sizeof(tlv));
+    }
+
+    return Send(aBuf, aLength);
+}
+
+Error BleSecure::Flush(void)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsConnected(), error = kErrorInvalidState);
+    VerifyOrExit(mSentMessage != nullptr, error = kErrorNoBufs);
+    VerifyOrExit(mSentMessage->GetLength() != 0, error = kErrorNone);
+
+    mTransmitQueue.Enqueue(*mSentMessage);
+    mTransmitTask.Post();
+
+    mSentMessage = Get<MessagePool>().Allocate(Message::kTypeBle);
+    VerifyOrExit(mSentMessage != nullptr, error = kErrorNoBufs);
```

Comment:
fixed.

Reply:
@abtink was this comment resolved already? Unfortunately I can't find back anymore which function this was related to since the line numbers have changed.